add currently known veYFI specs (#218)
yearn/yearn-devdocs: docs/contributing/governance/veyfi.md

# veYFI

> > This product is not deployed yet and is still under construction! You can follow development at https://github.com/yearn/veYFI

Author: @saltyfacu do you think merging this PR would be an issue even with this disclaimer that the product is not live yet? Just pinging to be sure this PR stays on-hold :)

Reviewer: yeah, I'd like it to be finished. We can coord with the team building it. I'd also like someone else to review this for accuracy

# veYFI

> This product is not deployed yet and is still under construction! You can follow development at https://github.com/yearn/veYFI

veYFI incorporates [YIP-56: Buyback and Build](https://gov.yearn.finance/t/yip-56-buyback-and-build/8929) funds into YFI tokenomics: Users can lock YFI tokens and receive veYFI which allows them to boost vault rewards and vote on where bought-back YFI will be sent to.

### Table-of-Content
- [Specification](#specification)
- [Benefits](#benefits)
- [Risks](#risks)

## Specification

### veYFI

- Receives bought-back YFI as rewards
- Locking similar to the ve-style program of Curve.

> - YFI can be locked up to 4 years into veYFI, which is non-transferable.

Reviewer 2: You say it can be locked up to 4 years, then right below it you say it hasn't been determined. I would consolidate the two points into 1.

> - The maximum lock duration is still to be determined, but will be in the range of min 1 year, max 4 years.

Reviewer 3: Min is one week

> - The maximum lock duration is still tbd, but will be in the range of min 1 year, max 4 years.

Reviewer 2: ```suggestion
- The maximum lock duration is still tbd, but will be in the range of min 1 year, max 4 years.
```
I wouldnt use shortened terms like this in the docs, best to write it out 'to be determined'

- Once veYFI is introduced, only veYFI is accepted voting power in Yearn Governance.

> ### Vault gauges + Voting

Reviewer 2: ```suggestion
### Vault Gauges + Voting
```
Capitalize titles / headings

- It’s possible to exit the lock early, in exchange for paying a penalty that gets allocated to the other veYFI holders.

> - Penalty size may be fixed (i.e. 50%), or may be depending on the remaining lock duration.

Reviewer 3: The penalty is up to 75% and decaying overtime. max(75%, time remaining / 4 years)

- Once veYFI is introduced, only veYFI is accepted voting power in Yearn Governance.

### Vault gauges + Voting

- Vault gauges allow vault depositors to stake their vault tokens and earn YFI rewards according to their veYFI weight.

> - YFI is allocated to gauges based on weekly governance votes. Each gauge can get a different amount of bought back YFI to emit.

Reviewer 3: bi-weekly

> - Based on their veYFI lock, users can boost their rewards of up to 2.5x proportional to the amount of vault tokens deposited, when they claim YFI rewards from gauges. The greater the amount of veYFI, the more vault deposits can be boosted for the user.

Reviewer 3: up to 10x

> - Inspired by Andre Cronje’s [initial design of Fixed Forex](https://andrecronje.medium.com/fair-launches-decentralized-collaboration-and-fixed-forex-ab327a2e4fc4), in order for gauge rewards to be claimed, the user must have a veYFI lock. Depending on their lock duration, they are entitled to a different share of gauge rewards:

Reviewer 3: That and the following lines should be removed, it's not the case anymore.
Instead: A claim with a boost under 100% will send the leftover boost to the veYFI holders.

  - if max lock = 4 years, and user is locked for 4 years, they are entitled to 100% of their rewards

>   - if user is locked for 2 years = 50% of rewards

Reviewer 2: of their rewards

>   - if user has no lock = 0% of their rewards

Reviewer 2: ```suggestion
  - if user has no lock = 0% of their rewards
```
maybe just say no rewards

> - You will be able to start claiming from the veFYI reward pool two or three weeks from the Thursday after which you lock before you can claim.

Reviewer 2: ```suggestion
- two or three weeks.
```
Hopefully we can get a definite number here before we merge this to the docs

> - You will be able to start claiming from the veFYI reward pool two or three weeks from the Thursday after which you lock before you can claim.

Author: ```suggestion
- You will be able to start claiming from the veYFI reward pool two or three weeks (to be defined) from the Thursday after which you lock before you can claim.
```

### Diagram

![](https://raw.githubusercontent.com/yearn/YIPS/master/YIPS/assets/yip-65/03-gauges.svg)

## Benefits

- **Incorporates YFI buybacks.** The mandate of [YIP-56: Buyback and Build](https://gov.yearn.finance/t/yip-56-buyback-and-build/8929) is unchanged, the new design builds on top of and integrates the bought back YFI.
- **It's a sustainable ecosystem.** The new design does not create a drain on Treasury assets. Instead, there are reinforcing flywheel effects where tokenomics rewards drive more TVL, that in turn drive more fees, that in turn drive more YFI buybacks, that is then used to reinforce the tokenomics.

> - **Incentivizes a **long-term** view on** Yearn.** Token holders are motivated to support the protocol over the long-term rather than to speculate in the short-term.

Reviewer 2: ```suggestion
- **Incentivizes a **long-term** view on** Yearn.** Token holders are motivated to support the protocol over the long-term rather than to speculate in the short-term.
```

The bold here is messed up
